shared/src/types/storage.rs

```diff
@@ -185,7 +185,13 @@ impl From<DbKeySeg> for Key {
         }
     }
 }
+impl FromStr for Key {
```

Comment:
(just a formatting nit, sadly rustfmt doesn't do this)
```suggestion

impl FromStr for Key {
```